### PR TITLE
Use compiler-local target environment

### PR DESCRIPTION
**Pull request by dsouzai**
Issue: #8198

Note, this PR hasn't removed all uses of the global TR::CompilerEnv->target

### CODE REVIEW

**Comment by dsouzai** [comment]
jenkins test sanity all jdk8

**Comment by dsouzai** [comment]
I just aborted the PR build because I just realized that it's 2020 and all the copyright checks are gonna fail :(

**Comment by dsouzai** [comment]
jenkins test sanity all jdk8

File: runtime/compiler/codegen/CodeGenGC.cpp [hunk 1]
```diff
@@ -378,7 +378,7 @@ J9::CodeGenerator::createStackAtlas()
             localObjectsFound = true;
             int32_t localObjectAlignment = comp->fej9()->getLocalObjectAlignmentInBytes();
             if (localObjectAlignment > stackSlotSize &&
-                (TR::Compiler->target.cpu.isX86() || TR::Compiler->target.cpu.isPower() || TR::Compiler->target.cpu.isZ()))
+                (comp->target().cpu.isX86() || comp->target().cpu.isPower() || comp->target().cpu.isZ()))
```
Comment (fjeremic) [inline runtime/compiler/codegen/CodeGenGC.cpp]: @0xdaryl @knn-k FYI I'm not sure why we have architecture guards for this. I'm sure ARM/AArch64 could benefit.

File: runtime/compiler/optimizer/IdiomTransformations.cpp [hunk 2]
```diff
@@ -4712,7 +4712,7 @@ makeCopyingTRTOInduction1Graph(TR::Compilation *c, int32_t ctrl, int32_t pattern
    tgt->setMinCounts(1, 1, 1);  // minimum ifCount, indirectLoadCount, indirectStoreCount
    tgt->setHotness(warm, false);
    static char *versionLengthStr = feGetEnv("TR_CopyingTRTOInduction1Graph_versionLength");
-   static int   versionLength = versionLengthStr ? atoi(versionLengthStr) : (TR::Compiler->target.cpu.isPower() ? 0 : 8);
+   static int   versionLength = versionLengthStr ? atoi(versionLengthStr) : (c->target().cpu.isPower() ? 0 : 8);
    tgt->setVersionLength(versionLength);   // depending on each architecture
```
Comment (fjeremic) [inline runtime/compiler/optimizer/IdiomTransformations.cpp]: This is pretty scary looking code in idiom recognition...

File: runtime/compiler/p/codegen/J9CodeGenerator.cpp [hunk 3]
```diff
@@ -728,9 +728,9 @@ J9::Power::CodeGenerator::insertPrefetchIfNecessary(TR::Node *node, TR::Register
       }
    else if (node->getOpCodeValue() == TR::awrtbari &&
             comp()->getMethodHotness() >= scorching &&
-            TR::Compiler->target.cpu.id() >= TR_PPCp6 &&
-              (TR::Compiler->target.is32Bit() ||
-                (TR::Compiler->target.is64Bit() &&
+            self()->comp()->target().cpu.id() >= TR_PPCp6 &&
+              (self()->comp()->target().is32Bit() ||
+                (self()->comp()->target().is64Bit() &&
                  comp()->useCompressedPointers() &&
```
Comment (fjeremic) [inline runtime/compiler/p/codegen/J9CodeGenerator.cpp]: Do these have to use `self()`? The one right below just calls `comp()`

Comment (dsouzai) [reply runtime/compiler/p/codegen/J9CodeGenerator.cpp]: 🤷‍♂ not sure, in an extensible class I always use `self()` just in case.

**Comment by fjeremic** [comment]
@dsouzai can you resolve the merge conflict?

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by dsouzai** [comment]
Conflict again...

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Comment by dsouzai** [comment]
@fjeremic all tests passed; do you mind merging

**Pull request merged**